### PR TITLE
BA-1808 [BE] Add basic profile information to JWT claims

### PR DESCRIPTION
**Pull request by tsl-ps2**
This adds basic profile information (id, name, image, urlPath) to the JWT claims

## Release Notes

- **New Features**
	- Introduced a new `JWTProfileSerializer` for improved user data serialization in JWT tokens.
	- Updated user profile retrieval to ensure unique results by eliminating duplicates.

- **Bug Fixes**
	- Adjusted the JWT token structure to include user names as part of a nested profile object.

- **Tests**
	- Enhanced test cases to reflect changes in user data structure, ensuring proper validation of profiles.

- **Chores**
	- Incremented version number from 0.4.9 to 0.4.10.

### CODE REVIEW

File: baseapp-auth/baseapp_auth/rest_framework/jwt/serializers.py [hunk 1]
```diff
@@ -26,13 +26,12 @@ def get_claim_serializer_class(cls):
 class BaseJwtLoginSerializer(
     CustomClaimSerializerMixin, LoginPasswordExpirationMixin, TokenObtainPairSerializer
 ):
-    @classmethod
-    def get_token(cls, user):
+    def get_token(self, user):
```
Comment (pt-tsl) [inline baseapp-auth/baseapp_auth/rest_framework/jwt/serializers.py:29]: Why did we remove the classmethod decorator here?

**Review by anicioalexandre — APPROVED** [review]

File: baseapp-auth/baseapp_auth/rest_framework/users/serializers.py [hunk 2]
```diff
@@ -14,13 +17,37 @@
...
 
 from baseapp_auth.password_validators import apply_password_validators
 
 
+class JWTProfileSerializer(serializers.ModelSerializer):
+    """
+    Serializes minimal profile data that will be attached to the JWT token as claim
+    """
+
+    url_path = serializers.SlugField(required=False, allow_null=True)
```
Comment (nossila) [inline baseapp-auth/baseapp_auth/rest_framework/users/serializers.py:30]: `url_path` is not a slug field, its a `ForeignKey`, and in the current expected behavior can contain `/` in it. Would be very good to make tests related to this.

File: baseapp-auth/baseapp_auth/rest_framework/jwt/serializers.py (hunk 1 above)
Comment (nossila) [reply baseapp-auth/baseapp_auth/rest_framework/jwt/serializers.py:29]: he is passing `self.context` to the serializer, that wouldn't be possible with a class method

File: baseapp-profiles/baseapp_profiles/managers.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 class ProfileManager(models.Manager):
     def filter_user_profiles(self, user):
-        return self.filter(models.Q(members__user=user) | models.Q(owner=user))
+        return self.filter(models.Q(members__user=user) | models.Q(owner=user)).distinct()
```
Comment (nossila) [inline baseapp-profiles/baseapp_profiles/managers.py:6]: this distincts are not needed, only if we have "bad" data, where the same user is both owner and member, but I'm ok with adding them, don't hurt.

File: baseapp-auth/baseapp_auth/rest_framework/users/serializers.py (hunk 2 above)
Comment (nossila) [reply baseapp-auth/baseapp_auth/rest_framework/users/serializers.py:30]: @tsl-ps2 we need to check this

**Review by anicioalexandre — APPROVED** [review]

File: baseapp-auth/baseapp_auth/rest_framework/jwt/serializers.py (hunk 1 above)
Comment (tsl-ps2) [reply baseapp-auth/baseapp_auth/rest_framework/jwt/serializers.py]: Reverted

File: baseapp-auth/baseapp_auth/rest_framework/users/serializers.py (hunk 2 above)
Comment (tsl-ps2) [reply baseapp-auth/baseapp_auth/rest_framework/users/serializers.py]: I replaced the `SlugField` by a `SerializerMethodField`

**Pull request merged**